src/WCSimLIGen.cc

```diff
@@ -0,0 +1,282 @@
+#include "WCSimLIGen.hh"
+#include "WCSimPrimaryGeneratorMessenger.hh"
+#include "WCSimPrimaryGeneratorAction.hh"
+
+#include "json.hpp"
+
+#include "G4ParticleGun.hh"
+#include "G4PhysicalConstants.hh"
+#include "G4OpticalPhoton.hh"
+#include "G4RandomDirection.hh"
+#include "G4SystemOfUnits.hh"
+
+using namespace std;
+using json = nlohmann::json;
+
+WCSimLIGen::WCSimLIGen(){
+
+    // Initialise
+    //myDetector = myDC;
+    //myMessenger = new WCSimLIGenMessenger(this);
+    this->Initialise();
+
+}
+
+WCSimLIGen::~WCSimLIGen(){
+
+    // things to delete?
+    //this->Finalise();
+}
+
+
+void WCSimLIGen::Initialise(){
+
+    // myMessenger, myDetector
+    photonMode = 0;
+    wcsimdir = string(getenv("WCSIMDIR"))+"/data/";
+} 
+
+void WCSimLIGen::SetPhotonMode(G4bool photonmode){
+   
+    if (photonmode){
+        photonMode=1;
+    }
+}
+    
+
+void WCSimLIGen::ReadFromDatabase(G4String injectorType, G4String injectorIdx, G4String injectorFilename){
+
+    // Define the database to read from
+    string db = injectorFilename;
+    if (db.empty()) {
+        db = wcsimdir + "LightInjectors.json";
+    }
+    else {
+        db = wcsimdir + injectorFilename;
+    }
+    
+    ifstream fJson(db.c_str());
+    
+    if (!fJson) {
+        G4cout << "LIGen: [ERROR] light injector db " << db << " not found, reverting to generic db" << G4endl;
```

Comment:
```suggestion
        G4cerr << "LIGen: [ERROR] light injector db " << db << " not found, reverting to generic db" << G4endl;
```

But I'm not sure this actually is "reverting" to anything, as `fJson` is read into `buffer` whether this is true or false